PowerEditor/src/ScitillaComponent/SmartHighlighter.cpp

```diff
@@ -56,27 +56,34 @@ void SmartHighlighter::highlightView(ScintillaEditView * pHighlightView)
 	char * text2Find = new char[textlen];
 	pHighlightView->getSelectedText(text2Find, textlen, false);	//do not expand selection (false)
 
+	
+	//GETWORDCHARS for isQualifiedWord2() and isWordChar2()
+	int listCharSize = pHighlightView->execute(SCI_GETWORDCHARS, 0, 0);
+	char *listChar = new char[listCharSize+1];
+	pHighlightView->execute(SCI_GETWORDCHARS, 0, (LPARAM)listChar);

```

Comment:
String returned by SCI_GETWORDCHARS from scintilla is not null terminated, so check for strlen in isWordChar() below on listChar is dangerous

- seen with MS Application Verifier on x64